### ISSUE TITLE
Incremental xDS

### ISSUE THREAD

**Issue by kyessenov** [istio/istio#6299]
Ref: https://github.com/envoyproxy/envoy/pull/3470

We would like envoy to dynamically request envoy resources to enable dynamic behavior controlled through the policy engine. Policy engine can issue route directives that allow envoy to divert traffic based on policy backend decisions. However, the control is limited by what the proxy already knows about (which is mostly inferred by envoy management without knowing any policies). Incremental xDS allows the proxy to dynamically fetch missing resources, expanding the space of possible routing changes. These resources have distinct availability properties from statically-provided resources.

This is pending on implementation of xDS incremental protocol in upstream envoy.

**Comment by andraxylia** [comment]
@dmitri-d yes, it is available, help is greatly appreciated!

**Comment by hzxuzhonghu** [comment]
Is there any plan to continue on this? I've noticed it is done on envoy side.

**Comment by seflerZ** [comment]
I think this is worthing especially for large scale clusters. We'll investigate.

**Comment by howardjohn** [comment]
Duplicate of https://github.com/istio/istio/issues/19681

This one was around first but the other has more conversation
